pkg/vault/contracts/VaultExtension.sol

```diff
@@ -180,6 +179,10 @@ contract VaultExtension is IVaultExtension, VaultCommon, Proxy {
             revert PoolAlreadyRegistered(pool);
         }
 
+        if (_poolTokens[pool].length > 0) {
+            revert TokensAlreadyRegistered(_poolTokens[pool]);
+        }
+
```

Comment:
I don't think we need another error here. How could this happen? We only add tokens in one place here, and there's no non-reverting path through this function where register could not get set.

If we wanted to be extra paranoid and do the check anyway, it's really the same as PoolAlreadyRegistered, so it would be:

```
    if (_isPoolRegistered(pool) || _poolTokens[pool].length > 0) {
        revert PoolAlreadyRegistered(pool);
    }
```
If we do that, maybe add tokens to the error: `revert PoolAlreadyRegistered(pool, _poolTokens[pool]);`

Reply:
I'd agree with merging this check as @EndymionJkb suggests.